src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.cs

```diff
@@ -137,5 +138,373 @@ namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : S
+                },
+                expectedOutput: !ExecutionConditionUtil.IsMonoOrCoreClr ? null : "1234",
+                verify: !ExecutionConditionUtil.IsMonoOrCoreClr ? Verification.Skipped : Verification.Passes);
+        }
+
+        [Fact]
+        public void MultipleInitializers_MultipleFiles()
+        {
+            string source1 = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C1
+{
+    [ModuleInitializer]
+    internal static void M1() => Console.Write(1);
+    [ModuleInitializer]
+    internal static void M2() => Console.Write(2);
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            string source2 = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C2
+{
+    internal class C3
+    {
+        [ModuleInitializer]
+        internal static void M3() => Console.Write(3);
+    }
+
+    [ModuleInitializer]
+    internal static void M4() => Console.Write(4);
+}
+
+class Program 
+{
+    static void Main() => Console.Write(6);
+}
+";
+
+            string source3 = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C4
+{
+    // shouldn't be called
+    internal static void M() => Console.Write(0);
+
+    [ModuleInitializer]
+    internal static void M5() => Console.Write(5);
+}
+";
+
+            CompileAndVerify(
+                new[] { source1, source2, source3 },
+                parseOptions: s_parseOptions,
+                options: TestOptions.DebugExe.WithMetadataImportOptions(MetadataImportOptions.All),
+                symbolValidator: module =>
+                {
+                    var rootModuleType = (TypeSymbol)module.GlobalNamespace.GetMember("<Module>");
+                    var staticConstructor = (PEMethodSymbol)rootModuleType.GetMember(".cctor");
+
+                    Assert.NotNull(staticConstructor);
+                    Assert.Equal(MethodKind.StaticConstructor, staticConstructor.MethodKind);
+
+                    var expectedFlags =
+                        MethodAttributes.Private
+                        | MethodAttributes.Static
+                        | MethodAttributes.SpecialName
+                        | MethodAttributes.RTSpecialName
+                        | MethodAttributes.HideBySig;
+
+                    Assert.Equal(expectedFlags, staticConstructor.Flags);
+                },
+                expectedOutput: "123456");
+        }
+
+        [Fact]
+        public void NonNullableField_01()
+        {
+            const string text = @"
+#nullable enable
+
+using System.Runtime.CompilerServices;
+
+class C
+{
+    static string s1;
+
+    [ModuleInitializer]
+    static void Init()
+    {
+        s1 = """";
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+
+            var comp = CreateCompilation(text, parseOptions: s_parseOptions);
+            comp.VerifyDiagnostics(
```

Comment:
>VerifyDiagnostics [](start = 17, length = 17)

VerifyEmitDiagnostics? Or better execute and observe if this scenario is really interesting.